File: packages/jexlang-java/src/main/java/com/jexlang/java/eval/errors/UndefinedVariableError.java
package com.jexlang.java.eval.errors;

public class UndefinedVariableError extends JexLangRuntimeError {
    private final String variableName;

    public UndefinedVariableError(String variableName) {
        super("Undefined variable: " + variableName);
        this.variableName = variableName;
    }

    public String getVariableName() {
        return variableName;
    }
}

File: packages/jexlang-swift/jexlang-swift/eval/errors/UndefinedVariableError.swift
//
//  UndefinedVariableError.swift
//  jexlang-swift
//
//  Created by Vinay Kumar on 03/10/25.
//

import Foundation

public class UndefinedVariableError: JexLangRuntimeError {
    public let variableName: String
    
    init(variableName: String) {
        self.variableName = variableName
        super.init(message: "Undefined variable: " + variableName)
    }
    
    public func getVariableName() -> String {
        return self.variableName;
    }
}
